### ISSUE TITLE
`node:` prefix not working with some rules

### ISSUE THREAD

**Issue by thernstig** [eslint-community/eslint-plugin-n#81]
I tested this using `"n/prefer-promises/fs": "error"` and it gave no error:

```js
import { readFile } from 'node:fs';

readFile('/etc/passwd', (err, data) => {
  if (err) throw err;
  console.log(data);
});
```

The rules `prefer-promises` does not seem to be able to handle using the `node:` prefix. Maybe this is true for several more rules?

**Comment by B2o5T** [comment]
just saw it also, any updates on it? I guess fixing it should be easy

**Comment by piranna** [comment]
I think https://github.com/eslint-community/eslint-plugin-n/pull/87 could fix this too.

**Comment by thernstig** [comment]
@piranna great! But it seems the referenced PR does not add any unit tests for that it works, which is probably important to not break it in the future.

**Comment by kevinoid** [comment]
[`n/no-deprecated-api`](https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-deprecated-api.md)  also appears to be affected.  For example, with

```js
const url1 = require('url');
const url2 = require('node:url');
url1.parse('http://example.com');
url2.parse('http://example.com');
```

An `n/no-deprecated-api` error occurs on line 3, but not line 4.
